src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -703,8 +720,13 @@ abstract class RefChecks extends Transform {
                           // TODO: what is the optimal way to test for a raw type at this point?
                           // Compilation has already failed so we shouldn't have to worry overmuch
                           // about forcing types.
-                          if (underlying.isJavaDefined && pa.typeArgs.isEmpty && abstractSym.typeParams.nonEmpty)
-                            ". To implement a raw type, use %s[_]".format(pa)
+                          if (underlying.isJavaDefined && pa.typeArgs.isEmpty && abstractSym.typeParams.nonEmpty) {
+                            prepareRawTypeForMessage(abstractSym).fold(
+                              ". To implement a raw type, use %s[_]".format(pa)
```

Comment:
I think this case should actually never happen. Am I right? Should that be an error instead of a default then? If yes, which?